pkg/plugin/config.go

```diff
@@ -30,6 +30,14 @@ type Config struct {
 	// event queue.
 	EventQueueWorkers int `json:"eventQueueWorkers"`
 
+	// StartupEventHandlingTimeoutSeconds gives the maximum duration, in seconds, that we are
+	// allowed to wait to finish handling all of the initial events generated by reading the cluster
+	// state on startup.
+	//
+	// If event processing takes longer than this time, then plugin creation will fail, and the
+	// scheduler pod will retry.
+	StartupEventHandlingTimeoutSeconds int `json:"startupEventHandlingTimeoutSeconds"`
```

Comment:
What does this timeout protects us against? 

If we have a lot of events on the start, maybe it is better to anyway process all of them? Otherwise, next initialization attempt will fail as well and plugin cannot start up. 

We can also not fail plugin's pod, and just keep on with inaccurate scheduling.

Reply:
The intent is that this should protect us against deadlocks, so that if event processing doesn't make enough process, we just restart.

Reply:
And how does it help? In case of a restart it will just keep trying to process the same events over and over again. 

But, if the event processing is not deadlocked, but just slow, it will take more time, but eventually the situation gets fixed.

Maybe instead of a restart it is better to fire an alert?

Reply:
I don't think we currently have a way for components to directly trigger an alert, at least not without metrics getting scraped. If the scheduler is down, someone will be paged either way - and we already have some alerts for if it's crashlooping.

I expect that restarting _may_ help against deadlocks because I _think_ it's more likely for us to have inconsistent deadlocks than it is for initial event processsing to just be really slow. But that's just a guess.

Reply:
> I don't think we currently have a way for components to directly trigger an alert

I meant, expose a metric startup_event_handling_seconds, and configure an alert when it is >30s.

> I expect that restarting may help against deadlocks because I think it's more likely for us to have inconsistent deadlocks than it is for initial event processsing to just be really slow. But that's just a guess.

If we were to have an alert:
 - If it is a deadlock, then by logs it should be possible to see if event processing is progressing or not, and a scheduler can be restarted manually
 - If it is slow, then the issue improves on it's own. 

Also, deadlock means a bug in the code, and writing specific protections for bugs seems like a grey path (why not just fix the bug?). 

But I guess we don't actually expect any of the issues to be happening regularly, and even we observe too many events, the mitigation is easy -> bump the timeout. So, leaving this up to you at this point.

Reply:
> expose a metric startup_event_handling_seconds

The tricky thing with this kind of metric is that we want it to keep updating even while we're waiting. AFAICT that's non-trivial to implement, so will keep everything as is for now, but I'd be happy to revisit this at some point in the future.